analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/EnvironmentalVariableScope.cs

```diff
@@ -0,0 +1,88 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    /// <summary>
+    /// Defines a scope inside which new environment variables can be set.
+    /// The variables will be cleared when the scope is disposed.
+    /// Copy-pasted from S4NET codebase.
+    /// </summary>
+    public sealed class EnvironmentVariableScope : IDisposable
+    {
+        private readonly bool setOnlyInRelease;
+        private IDictionary<string, string> originalValues = new Dictionary<string, string>();
+
```

Comment:
As an API, this class is general and therefore quite hard to use for it's main purpose.
Let's keep it's methods for advanced scenarios (like the `tRUE` case) and add writeonly property for each well-known-often-needed variable.
```
#pragma warning disable S2376 // Write-only properties should not be used

        public bool EnableConcurrentAnalysis
        {
            set
            {
                SetVariable(SonarDiagnosticAnalyzer.EnableConcurrentExecutionVariable, value.ToString());
            }
        }

#pragma warning restore S2376
```
that will enable simple usage like this on most of places
```
using var _ = new EnvironmentVariableScope { EnableConcurrentAnalysis = true };
```

Reply:
Let's use `using var _ =` everywhere instead of `using var scope =`. We don't need that variable.